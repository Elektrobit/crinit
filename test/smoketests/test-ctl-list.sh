# shellcheck shell=bash
# SPDX-License-Identifier: MIT
#
# smoketest for crinit-ctl list command
#

setup() {
    crinit_config_setup
}

run() {
    crinit_daemon_start "${SMOKETESTS_CONFDIR}"/demo.series
    sleep 3

    out="${SMOKETESTS_RESULTDIR}"/"${SMOKETESTS_NAME}".out
    sample="${CMDPATH}"/test-"${SMOKETESTS_NAME}".out
    "$BINDIR"/crinit-ctl list > "$out"
    compare_output "$sample" "$out"
}

teardown() {
    # Terminate crinit daemon
    crinit_daemon_stop
}

