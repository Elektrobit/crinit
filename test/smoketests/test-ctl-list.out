NAME                     PID  USER GROUP STATUS
hello_world               -1  root root  done
sleep_one_sec             -1  root root  done
after_sleep               -1  root root  done
one_second_respawn        -1  root root  loaded
fail_loop                 -1  root root  loaded
feature_provider          -1  root root  loaded
feature_dependent         -1  root root  loaded
stop_command              -1  root root  done
stop_command_two_lines    -1  root root  done
stop_command_var_exp      -1  root root  done
